ase/scripts/generate_ase_environment.py

```diff
@@ -282,6 +288,130 @@ def config_sources(fd, filelist):
     return json_file
 
 
+#
+# Qsys has a compilation step.  When using case #1 above, detect Qsys
+# sources, compile them, and include the generated Verilog in the
+# simulation.
+#
+def config_qsys_sources(filelist, vlog_srcs):
+    # Get all the sources.  rtl_src_config will emit all relevant source
+    # files, one per line.
+    try:
+        srcs = commands_list_getoutput(
+            "rtl_src_config --qsys --abs".split(" ") + [filelist])
+    except Exception:
+        errorExit("failed to read sources from {0}".format(filelist))
+
+    # Collect two sets: qsys source files and the names of directories into
+    # which generated Verilog will be written.  The directory names match
+    # the source names.
+    qsys_srcs = []
+    ip_dirs = []
+    srcs = srcs.split('\n')
+    for s in srcs:
+        if (s):
+            # Record all build target directories
+            ip_dirs.append(os.path.splitext(s)[0])
+
+            # Collect all qsys files
+            if (s.lower()[-5:] == '.qsys'):
```

Comment:
what is -5? If you intended to use the length of string ".qsys", can you use the python equivalent of strlen

Reply:
-5 is used to get the last five characters in string 's' in python.

Reply:
I just discovered Python's str.endswidth(). I'll switch to that.